sdk/python/kfp/components/_structures.py

```diff
@@ -90,6 +90,8 @@ def __init__(self,
         type: Optional[TypeSpecType] = None,
         description: Optional[str] = None,
         annotations: Optional[Dict[str, Any]] = None,
+        default: Optional[str] = None,
+        optional: Optional[bool] = False,
```

Comment:
I'm not sure we want to support optional outputs at this moment. The producer component itself can implement the default value logic if needed.